packages/affiliates/libs/models.js

```diff
@@ -171,13 +171,14 @@ function History() {
   const history = [];
   // Used internally, but will insert a block into cache sorted by timestamp ascending
   function insert(data) {
-    assert(data.blockNumber, "requires blockNumber");
+    assert(data.blockNumber === 0 || data.blockNumber > 0, "requires blockNumber");
```

Comment:
What is this testing? Not entirely self-documenting since it _feels_ like this could be reduced to blockNumber >= 0? Is it trying to ensure the type is a number too? Or that it's also not undefined? Might be easier to test that in a specific ||, like:
```
assert(typeof(data.blockNumber) === "number" && data.blockNumber >= 0, "requires blockNumber");
```

Or a comment would suffice -- I just find the == 0 || > 0 expression a little confusing.

Reply:
The goal was to allow 0 and above but not undefined or null, it was the best i could come up with, but yours i think i agree is easier to understand! Will update.